brainframe_qt/api_utils/streaming/synced_reader.py

```diff
@@ -1,177 +1,142 @@
+        super().__init__(parent=parent)
+
         self.stream_id = stream_id
         self._stream_reader = stream_reader
-        self.status_receiver = status_receiver
 
         self.latest_processed_frame: Optional[ZoneStatusFrame] = None
-        """Latest frame synced with results. None if no frames have been synced 
-        yet"""
+        """Latest frame synced with results. None if no frames have been synced yet"""
 
-        self.stream_listeners: Set[StreamListener] = set()
-        self._stream_listeners_lock = RLock()
+        self.frame_syncer = FrameSyncer()
 
-        # Start threads, now that the object is all set up
-        self._thread = Thread(
-            name=f"SyncedStreamReader thread for stream ID {stream_reader}",
-            target=self._sync_detections_with_stream,
-            daemon=True
-        )
-        self._thread.start()
-
-    def alert_frame_listeners(self):
-        with self._stream_listeners_lock:
-            if self.status is StreamStatus.STREAMING:
-                for listener in self.stream_listeners:
-                    listener.frame_event.set()
-
-    def alert_status_listeners(self, status: StreamStatus):
-        """This should be called whenever the StreamStatus has changed"""
-        with self._stream_listeners_lock:
-            if status is StreamStatus.INITIALIZING:
-                for listener in self.stream_listeners:
-                    listener.stream_initializing_event.set()
-            elif status is StreamStatus.HALTED:
-                for listener in self.stream_listeners:
-                    listener.stream_halted_event.set()
-            elif status is StreamStatus.CLOSED:
-                for listener in self.stream_listeners:
-                    listener.stream_closed_event.set()
-            else:
-                logging.critical("SyncedStreamReader: An event was called, but"
-                                 " was not actually necessary!")
-                for listener in self.stream_listeners:
-                    listener.stream_error_event.set()
-
-    def add_listener(self, listener: StreamListener):
-        with self._stream_listeners_lock:
-            self.stream_listeners.add(listener)
-            if self.status is not StreamStatus.STREAMING:
-                self.alert_status_listeners(self.status)
-            elif self.latest_processed_frame is not None:
-                self.alert_frame_listeners()
-            else:
-                listener.stream_initializing_event.set()
+    def run(self) -> None:
+        while self.status is not StreamStatus.INITIALIZING:
+            sleep(0.01)
+
+        frame_or_status_event = or_events(self._stream_reader.new_frame_event,
+                                          self._stream_reader.new_status_event)
+
+        while (
+            self.status is not StreamStatus.CLOSED
+            and not self.thread().isInterruptionRequested()
+        ):
+            if not frame_or_status_event.wait(10):
```

Comment:
Is this waiting for 10 seconds? I assume that means this thread could take up to ten seconds to close out, which seems long. Maybe this should be reduced to something like one second.

Reply:
Oh I totally thought this was ms. Will change